### ISSUE TITLE
fix: add git would fail if path is null

### ISSUE THREAD

**Issue by huang12zheng** [felangel/mason#163]
**Description**
```dart
 void _ensureRemoteBrickExists(Directory directory, GitPath gitPath) {
      final brickYaml = File(
        p.join(directory.path, gitPath.path, BrickYaml.file),
      );
      if (!brickYaml.existsSync()) {
        directory.deleteSync(recursive: true);
        throw BrickNotFoundException('${gitPath.url}/${gitPath.path}');
      }
    }
```
if path is null, it would error.

https://github.com/felangel/mason/blob/master/lib/src/bricks_json.dart#:~:text=directory%2C%20GitPath%20gitPath)%20%7B-,final%20brickYaml%20%3D%20File(,)%3B,-if%20(!brickYaml.existsSync

**reproduction**
mason add -g --source git https://github.com/RomanticEra/bricks

**Expected Behavior**
if path is null , replace it ''

**Comment by huang12zheng** [comment]
```dart
p.join(directory.path, gitPath.path ?? '', BrickYaml.file),
```
just a  suggestion, it would work.

**Comment by huang12zheng** [comment]
![截屏2021-12-11 上午12 18 45 (2)](https://user-images.githubusercontent.com/28038074/145606487-1342e51f-b779-4d57-a5a0-bc5ea26fc3ad.png)

Because there are only three places, Maybe it would be good to change it as follow
```dart
const GitPath(this.url, {this.path='', this.ref});
```

or change that three places.

What's your suggestion?
@felangel

**Comment by felangel** [comment]
This has been resolved as part of #162
